cmd/jaeger/internal/extension/jaegerstorage/extension_test.go

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package jaegerstorage
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/extension"
+	noopmetric "go.opentelemetry.io/otel/metric/noop"
+	nooptrace "go.opentelemetry.io/otel/trace/noop"
+	"go.uber.org/zap"
+
+	memoryCfg "github.com/jaegertracing/jaeger/pkg/memory/config"
+	"github.com/jaegertracing/jaeger/pkg/metrics"
+	"github.com/jaegertracing/jaeger/storage"
+	"github.com/jaegertracing/jaeger/storage/dependencystore"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+)
+
+const memstoreName = "memstore"
+
+type storageHost struct {
+	t                *testing.T
+	storageExtension component.Component
+}
+
+func (host storageHost) GetExtensions() map[component.ID]component.Component {
+	return map[component.ID]component.Component{
+		ID: host.storageExtension,
+	}
+}
+
+func (host storageHost) ReportFatalError(err error) {
+	host.t.Fatal(err)
+}
+
+func (storageHost) GetFactory(_ component.Kind, _ component.Type) component.Factory {
+	return nil
+}
+
+func (storageHost) GetExporters() map[component.DataType]map[component.ID]component.Component {
+	return nil
+}
+
+type errorFactory struct {
+	closeErr error
+}
+
+func (e errorFactory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger) error {
+	panic("implement me")
+}
+
+func (e errorFactory) CreateSpanReader() (spanstore.Reader, error) {
+	panic("implement me")
+}
+
+func (e errorFactory) CreateSpanWriter() (spanstore.Writer, error) {
+	panic("implement me")
+}
+
+func (e errorFactory) CreateDependencyReader() (dependencystore.Reader, error) {
+	panic("implement me")
+}
+
+func (e errorFactory) Close() error {
+	return e.closeErr
+}
+
+func TestExtensionConfigError(t *testing.T) {
+	config := createDefaultConfig().(*Config)
+	err := config.Validate()
+	require.EqualError(t, err, fmt.Sprintf("%s: no storage type present in config", ID))
+}
+
+func TestStartStorageExtensionTwiceError(t *testing.T) {
+	ctx := context.Background()
+
+	storageExtension := makeStorageExtension(t, memstoreName)
+
+	host := componenttest.NewNopHost()
+	err := storageExtension.Start(ctx, host)
+	require.Error(t, err)
+	require.EqualError(t, err, fmt.Sprintf("duplicate memory storage name %s", memstoreName))
+}
+
+func TestBadHostGetStorageFactoryError(t *testing.T) {
+	makeStorageExtension(t, memstoreName)
+
+	host := componenttest.NewNopHost()
+	_, err := GetStorageFactory(memstoreName, host)
+	require.Error(t, err)
+	require.EqualError(t, err, fmt.Sprintf("cannot find extension '%s' (make sure it's defined earlier in the config)", ID))
+}
+
+func TestBadNameGetStorageFactoryError(t *testing.T) {
```

Comment:
nit: the test names are meant to be hierarchical {componentName}{testScenario}. I.e. everything here should have the prefix TestStorageFactory***